File: Dwd.Lab.Dominio/Entidades/Produto.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace Dwd.Lab.Dominio.Entidades
{
    public class Produto
    {
        public int Id { get; set; }
        public int CategoriaProdutoId { get; set; }
        
        public int? SubCategoriaProdutoId { get; set; }

        public string CodigoLoja { get; set; }
        public string Nome { get; set; }
        public string Descricao { get; set; }
        public double Preco { get; set; }
        public virtual CategoriaProduto CategoriaProduto { get; set; }
        public virtual SubCategoriaProduto SubCategoriaProduto { get; set; }


    }
}

File: Dwd.Lab.Web/ClientApp/src/app/Models/produto.ts
import { CategoriaProduto } from "./categoria-produto";
import { SubCategoriaProduto } from "./sub-categoria-produto";

export class Produto {
    id: number;
    codigoLoja: string;
    nome: string;
    descricao: string;
    preco: number;
    categoriaProduto: CategoriaProduto;
    subCategoriaProduto: SubCategoriaProduto;

}